include/mxnet/resource.h

```diff
@@ -148,33 +153,38 @@ struct Resource {
    * \brief Get space requested as mshadow Tensor in specified type.
    *  The caller can request arbitrary size.
    *
-   * \param shape the Shape of returning tensor.
-   * \param stream the stream of retruning tensor.
+   * \param shape  the Shape  of returning tensor.
+   * \param stream the Stream of returning tensor.
+   * \parma name   the Name of the operator requesting the resource.
    * \return the mshadow tensor requested.
    * \tparam xpu the device type of random number generator.
    * \tparam ndim the number of dimension of the tensor requested.
    */
   template<typename xpu, int ndim, typename DType>
   inline mshadow::Tensor<xpu, ndim, DType> get_space_typed(
-      mshadow::Shape<ndim> shape, mshadow::Stream<xpu> *stream) const {
+      mshadow::Shape<ndim> shape, mshadow::Stream<xpu> *stream,
+      const std::string &name =
+        MXNET_STORAGE_DEFAULT_NAME_FARG("temp_space")) const {
```

Comment:
should the name include where the allocation happened (e.g. GPU/stream)?

Reply:
I agree that we should include the GPU, but for stream I think we need to discuss.

Reply:
Let's just add GPU and leave out the stream for now